### PR TITLE
Proposal for varios changes 

### PR DESCRIPTION
**Pull request by helicocrasher**
Hi @Tjoms99 

So as propsed lets try a pull request. It is the first one I ever tried, so you might need to be patient.

Main Changes:
- Setting of the BQ25101 Chip. On function to set the charge current (100 mA 50mA or 0 mA)
- elimination of the enable routines as they do no really work with the BQ25101 - instead status read out wheter charging is active is equal to the chargeLED.
- Voltage read during charging switches to 0mA to be able to read battery voltage .. and after reading restores the previousls se charge current.
- LED blinking blue green red off just for the fun of it. 

Please check which parts you might like.

Cheers

### CODE REVIEW

**Comment by Tjoms99** [comment]
Good stuff! Will test the code later, left some comments here for you to think about:) Hope this wasnt too harsh

**Comment by Tjoms99** [comment]
Hi @helicocrasher, it has been a while. If you would like some help with the feedback, let me know!

File: src/battery/battery.h [hunk 1]
```diff
@@ -19,35 +19,24 @@
...
 #define __BATTERY_H__
 
-/**
- * @brief Set battery charging to fast charge (100mA).
- *
- * @retval 0 if successful. Negative errno number on error.
- */
-int battery_set_fast_charge(void);
+typedef enum batt_charging_status{disconnected=-1,notCharging=0,Charging=1} enum_bachast; 
+typedef enum batt_max_charge_current{Current0=0, Current50=50,Current100=100} enum_bachacurr;
```
[the remark this answers is not in the document]
Comment (helicocrasher) [reply src/battery/battery.h:23]: I am back ..

Thanks for you comments - good education. (I have not been writing SW for a long while)
So coding style and convention, I can learn a lot.

File: src/battery/battery.c [hunk 2]
```diff
@@ -22,15 +22,20 @@
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/adc.h>
 #include <zephyr/logging/log.h>
+
+//LOG_MODULE_REGISTER(battery, LOG_LEVEL_DBG|LOG_LEVEL_INF);
```
[the remark this answers is not in the document]
Comment (helicocrasher) [reply src/battery/battery.c:26]: sure  when put in a library commente out debug code has no place.

File: src/battery/battery.c [hunk 3]
```diff
@@ -99,47 +104,26 @@ static int battery_enable_read()
...
-{
-    if (!is_initialized)
-    {
-        return -ECANCELED;
-    }
-
-    return gpio_pin_set(gpio_battery_dev, GPIO_BATTERY_CHARGING_ENABLE, 0);
+void battery_get_charging_status(enum_bachast *battery_charging_status){
+    *battery_charging_status=gpio_pin_get(gpio_battery_dev, GPIO_BATTERY_CHARGING_STATUS);
+    LOG_DBG("battery_get_charging_satus (Status %d \n)", *battery_charging_status);
```
[the remark this answers is not in the document]
Comment (helicocrasher) [reply src/battery/battery.c:126]: thanks

**Comment by Tjoms99** [comment]
Idle, will close this issue

**Pull request closed without merging**